pkg/ali/components/CruACK.vue

```diff
@@ -0,0 +1,656 @@
+
+      return validVersions;
+    },
+
+    cancelCredential() {
+      if ( this.$refs.cruresource ) {
+        (this.$refs.cruresource).emitOrRoute();
+      }
+    },
+    onMembershipUpdate(update) {
+      this['membershipUpdate'] = update;
+    },
+    async saveRoleBindings() {
+      if (this.membershipUpdate.save) {
+        await this.membershipUpdate.save(this.normanCluster.id);
+      }
+    },
+    async actuallySave() {
+      await this.normanCluster.save();
+
+      return await this.normanCluster.waitForCondition('InitialRolesPopulated');
+    },
+    setClusterName(name) {
+      this.normanCluster['name'] = name;
+
+      if (!this.isImport) {
+        this.config['clusterName'] = name;
+      }
+    },
+    removePool(i) {
+      const pool = this.nodePools[i];
+
+      removeObject(this.nodePools, pool);
+    },
+
+    addPool() {
+      const poolName = `nodePool-${ this.nodePools.length + 1 }`;
+      const _id = randomStr();
+      const neu = {
+        ...cloneDeep(DEFAULT_NODE_GROUP_CONFIG), name: poolName, _id, _isNewOrUnprovisioned: true, version: this.config.kubernetesVersion
+      };
+
+      this.nodePools.push(neu);
+
+      this.$nextTick(() => {
+        const pools = this.$refs.pools;
+
+        if ( pools && pools.select ) {
+          pools.select(poolName);
+        }
+      });
+    },
+    async getAllInstanceTypes() {
+      this.loadingInstanceTypes = true;
+      const { alibabaCredentialSecret, regionId } = this.config;
+
+      try {
+        this.allInstanceTypes = {};
+        let nextToken;
+
+        do {
+          const res = await getAllAlibabaInstanceTypes(this.$store, alibabaCredentialSecret, regionId, nextToken);
+          const fromRes = res?.InstanceTypes?.InstanceType || [];
+
+          fromRes.forEach((type) => {
+            if (type.InstanceTypeId) {
+              this.allInstanceTypes[type.InstanceTypeId] = {
+                instanceTypeFamily: type.InstanceTypeFamily,
+                cpu:                type.CpuCoreCount,
+                memory:             type.MemorySize,
+              };
+            }
+          });
+          nextToken = res?.NextToken;
+        } while (nextToken);
+      } catch (err) {
+        const parsedError = err.error || '';
+
+        this.$emit('error', this.t('ack.errors.instanceTypes', { e: parsedError || err }));
+      }
+      this.loadingInstanceTypes = false;
+    },
+
+  }
+});
+</script>
+<template>
+  <Loading v-if="$fetchState.pending" />
+  <CruResource
+    v-else
+    ref="cruresource"
+    :resource="value"
+    :mode="mode"
+    :can-yaml="false"
+    :done-route="doneRoute"
+    :validation-passed="fvFormIsValid && configIsValid"
+    @error="e=>errors=e"
+    @finish="save"
+  >
+    <div
+      v-if="hasCredential"
+      class="row mb-20"
+    >
+      <div class="col span-4">
+        <LabeledInput
+          :value="normanCluster.name"
+          :mode="mode"
+          label-key="generic.name"
+          required
```

Comment:
this is marked as required but "Not Selected" is also an option. Is "Not Selected" a valid choice?